From c8ca9e2289f1fc47d07463b82371a23d3b670fe4 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Thu, 11 Apr 2024 23:43:39 +0200
Subject: [PATCH 1109/1141] ARM: dts: stm32: fix irq type of touchscreen on
 stm32mp135f-dk

The touchscreen driver goodix_ts ignores the interrupt type set in
the DT; it queries the touchscreen firmware for the interrupt type
to be requested.
When the touchscreen device is unbound, the interrupt keeps the
type already requested.
On a following bind of the same driver, the interrupt framework
will find that the interrupt is already registered and will check
if its interrupt type matches the DT; in case of type mismatch the
bind will fail.

The touchscreen firmware can be update during the product life,
thus the interrupt type can change.

Use IRQ_TYPE_NONE in the DT, as it works as a wildcard and matches
any interrupt type requested by the firmware.

Current touchscreen firmware in stm32mp135f-dk require falling
edge type. This already matches the type in DT.

Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: If0a3957a080f960db1977e5ffafe1d51475f9e57
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/373138
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Amelie DELAUNAY <amelie.delaunay@foss.st.com>
---
 arch/arm/boot/dts/stm32mp135f-dk.dts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/stm32mp135f-dk.dts b/arch/arm/boot/dts/stm32mp135f-dk.dts
index f30a5618a1d8..40e02fe88a40 100644
--- a/arch/arm/boot/dts/stm32mp135f-dk.dts
+++ b/arch/arm/boot/dts/stm32mp135f-dk.dts
@@ -409,7 +409,7 @@ goodix: goodix-ts@5d {
 		pinctrl-names = "default";
 		pinctrl-0 = <&goodix_pins_a>;
 		interrupt-parent = <&gpiof>;
-		interrupts = <5 IRQ_TYPE_EDGE_FALLING>;
+		interrupts = <5 IRQ_TYPE_NONE>;
 		reset-gpios = <&gpioh 2 GPIO_ACTIVE_LOW>;
 		AVDD28-supply = <&scmi_v3v3_sw>;
 		VDDIO-supply = <&scmi_v3v3_sw>;
-- 
2.39.2

